strawberry/tools/merge_types.py

```diff
@@ -0,0 +1,35 @@
+from collections import Counter
+from itertools import chain
+from typing import Tuple, Type
+
+import strawberry
+
+
+DEFAULT_NAME = "MegaType"
+
+
+def merge_types(types: Tuple[Type], name: str = DEFAULT_NAME) -> Type:
+    """Merge multiple Strawberry types into one
+
+    For example, given two queries `A` and `B`, one can merge them as follows:
+
+        merge_types((B, A))
+
+    This is essentially the same as:
+
+        class Query(B, A):
+            ...
+
+    An optional name may be specified for the resulting type.
+    """
+
+    if not types:
+        raise ValueError("Can't merge types if none are supplied")
+
+    fields = chain(*(t._type_definition.fields for t in types))
+    counter = Counter(f.name for f in fields)
+    dupes = [f for f, c in counter.most_common() if c > 1]
+    if dupes:
+        raise Warning("{} has overridden fields: {}".format(name, ", ".join(dupes)))
```

Comment:
I think we should raise an exception here instead of a "Warning" since we're stopping execution  (maybe a TypeError 🤷 ). In my mind warnings are informational and shouldn't stop code execution but I think this warrants being an actual error.

Reply:
Dynamic languages usually allow one to redeclare variables, override properties, methods etc. Programs still run, and do so with no side effects. Raising an error would stop program execution, which is both unwarranted as the program would still run just fine, and surprising as it goes against the rules of the language. A warning, on the other hand, makes no assumptions about user intent, but still does its duty.

Reply:
Ok sure that's fine but you are raising an error here which will stop program execution. Did you mean to log the warning instead?

Reply:
> Did you mean to log the warning instead?

😬 That's correct - see https://github.com/strawberry-graphql/strawberry/pull/1273/commits/7261ddbf7d6c0fabc4f1be5cb7d250a716b3476c

Reply:
Perfect 😃